nautobot/extras/models/groups.py

```diff
@@ -56,87 +57,232 @@ class DynamicGroup(OrganizationalModel):
+            if to_field_name is not None:
+                modelform_field.to_field_name = to_field_name
+
+            logger.debug("Added %s (%s) to filter fields", missing_field, modelform_field.__class__.__name__)
+            filterform_fields[missing_field] = modelform_field
+
+        # Reduce down to a final dict of desired fields.
+        return_fields = {}
+        for field_name, filter_field in filterform_fields.items():
+            # Skip excluded fields
+            if field_name.startswith(self.exclude_filter_fields):
+                logger.debug("Skipping excluded filter field: %s", field_name)
+                continue
+
+            return_fields[field_name] = filter_field
+
+        return return_fields
+
+    def get_filter_fields(self):
+        """Return a mapping of `{field_name: filter_field}` for this group's `content_type`."""
+        # Fail cleanly until the object has been created.
+        if self.model is None:
+            return {}
+
+        if not self.present_in_database:
+            return {}
+
+        return self._map_filter_fields
+
+    def get_queryset(self):
+        """
+        Return a queryset for the `content_type` model of this group.
+
+        The queryset is generated based on the `filterset_class` for the Model.
+        """
+
+        model = self.model
 
         if not self.filter:
             return model.objects.none()
 
-        qs = self.map.get_queryset(self.filter, **kwargs)
+        filterset = self.filterset_class(self.filter, model.objects.all())
+        qs = filterset.qs
 
-        # Make sure that this instance can't be a member of its own group
+        # Make sure that this instance can't be a member of its own group.
         if self.present_in_database and model == self.__class__:
             qs = qs.exclude(pk=self.pk)
 
         return qs
 
     @property
     def members(self):
+        """Return the member objects for this group."""
         return self.get_queryset()
 
     @property
     def count(self):
-        """Return the number of objects in the group."""
+        """Return the number of member objects in this group."""
         return self.get_queryset().count()
 
-    @property
-    def map(self):
-        """
-        Accessor to automatically generated `BaseDynamicGroupMap` class.
-
-        This class object is cached on the instance after the first time it is accessed.
-        """
-
-        if getattr(self, "_map", None) is None:
-            try:
-                model = self.content_type.model_class()
-                dynamicgroupmap_class = dynamicgroup_map_factory(model)
-            except (models.ObjectDoesNotExist, TypeError):
-                dynamicgroupmap_class = None
-
-            self._map = dynamicgroupmap_class
-
-        return self._map
-
     def get_absolute_url(self):
         return reverse("extras:dynamicgroup", kwargs={"slug": self.slug})
 
+    @property
+    def base_url(self):
```

Comment:
I'm still finding `base_url` to not be especially obvious in its meaning. Maybe something like `members_base_url` or the like might be clearer?